app/models/plan.rb

```diff
@@ -196,37 +196,70 @@ def get_guidance_group_options
   #  emails org admins and org contact 
   #  adds org admins to plan with the 'reviewer' Role
   def request_feedback(user)
-    val = Role.access_values_for(:reviewer, :commenter).min
-    self.feedback_requested = true
+    Plan.transaction do
+      begin
+        val = Role.access_values_for(:reviewer, :commenter).min
+        self.feedback_requested = true
     
-    # Share the plan with each org admin as the reviewer role
-    admins = user.org.org_admins
-    admins.each do |admin|
-      role = Role.new(user: admin, access: val)
-      self.roles << role unless self.users.include?(admin)
-    end 
-
-    if self.save!
-      # Send an email confirmation to the owners and co-owners
-      self.owner_and_coowners.each do |owner|
-        UserMailer.feedback_confirmation(owner, self, user).deliver_now
-      end
-  
-      # Send an email to all of the org admins as well as the Org's administrator email
-      if user.org.contact_email.present?
-        admins << User.new(email: user.org.contact_email, firstname: user.org.contact_name)
+        # Share the plan with each org admin as the reviewer role
+        admins = user.org.org_admins
+        admins.each do |admin|
+          self.roles << Role.new(user: admin, access: val)
+        end 
+
+        if self.save!
+          # Send an email confirmation to the owners and co-owners
+          deliver_if(recipients: self.owner_and_coowners, key: 'users.feedback_requested') do |r|
+            UserMailer.feedback_confirmation(r, self, user).deliver_now
+          end
+          # Send an email to all of the org admins as well as the Org's administrator email
+          if user.org.contact_email.present?
+            admins << User.new(email: user.org.contact_email, firstname: user.org.contact_name)
+          end
+          deliver_if(recipients: admins, key: 'admins.feedback_requested') do |r|
+            UserMailer.feedback_notification(r, self, user).deliver_now
+          end
+          true
+        else
+          false
+        end
+      rescue Exception => e
+        Rails.logger.error e
+        false
       end
+    end
+  end
 
-      deliver_if(recipients: admins, key: 'admins.feedback_requested') do |r|
-        UserMailer.feedback_notification(r, self, user).deliver_now
+  ##
+  # Finalizes the feedback for the plan:
+  #  emails confirmation messages to owners
+  #  sets flag on plans.feedback_requested to false
+  #  removes org admins from the 'reviewer' Role for the Plan
+  def complete_feedback(org_admin)
+    Plan.transaction do
```

Comment:
nice using here transaction to be safe about the feedback_requested flag.I guess is needed since multiple admins can review a plan simultaneously, otherwise it might be overkill...